doc/specs/#166 - Font Support.md

```diff
@@ -0,0 +1,387 @@
+| InstalledLocation    |   N/A    | %LOCALAPPDATA%/Microsoft/Windows/Fonts/Arial.ttf |
+| InstalledScope    |       |   User |
+
+> The unique identifer for each font will be a combination of the scope, font font family name, and font face name.
+
+> The version of the font only exists at the font face level and not at the family level
+
+## CLI Behavior / Implementation Details
+
+New Command: `winget fonts`
+---
+Fonts should be completely separate from the existing package management experience. Fonts aren't as complex as installing packages so having new subcommands can help simplify the necessary arguments for proper functionality. To support this, a new command called `font` will be added. The following subcommands will be available:
+- `winget fonts list`
+- `winget fonts install`
+- `winget fonts uninstall`
+- `winget fonts upgrade`
+- `winget fonts show`
+
+`winget fonts list`
+---
+The default behavior of `winget fonts list` is to display all installed font families and the number of faces that belong to each font family.
+
+| Family Name | Face Count |
+| -------- | ------- |
+| Arial    |   6    |
+| Bahnschrift | 1   |
+| Cascadia Code | 30 |
+
+```c++
+    // How to enumerate the list of installed font family names.
+    wchar_t localeNameBuffer[LOCALE_NAME_MAX_LENGTH];
+    const auto localeName = GetUserDefaultLocaleName(localeNameBuffer, LOCALE_NAME_MAX_LENGTH) ? localeNameBuffer : L"en-US";
+
+    wil::com_ptr<IDWriteFactory7> factory;
+    THROW_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(factory), factory.put_unknown()));
+
+    wil::com_ptr<IDWriteFontCollection> collection;
+    THROW_IF_FAILED(factory->GetSystemFontCollection(collection.addressof(), FALSE));
+
+    const auto familyCount = collection->GetFontFamilyCount();
+
+    for (UINT32 familyIndex = 0; familyIndex < familyCount; familyIndex++)
+    {
+        // Get font family from font collection by index.
+        wil::com_ptr<IDWriteFontFamily> family;
+        THROW_IF_FAILED(collection->GetFontFamily(familyIndex, family.addressof()));
+
+        // Get font family name.
+        wil::com_ptr<IDWriteLocalizedStrings> names;
+        THROW_IF_FAILED(family->GetFamilyNames(names.addressof()));
+
+        UINT32 index;
+        BOOL exists;
+        if (FAILED(names->FindLocaleName(localeName, &index, &exists)) || !exists)
+        {
+            index = 0;
+        }
+
+        UINT32 nameLength;
+        THROW_IF_FAILED(names->GetStringLength(index, &nameLength));
+        nameLength += 1; // Account for the trailing null terminator during allocation.
+
+        wchar_t name[512];
+        const auto fontCount = family->GetFontCount();
+        THROW_HR_IF(E_OUTOFMEMORY, nameLength > ARRAYSIZE(name));
+        THROW_IF_FAILED(names->GetString(0, &name[0], nameLength));
+    }
+```
+
+If a user specifies a specific name such as : 
+
+`winget fonts list --name 'Arial'`
```

Comment:
Will this support a substring match like `winget list`? Can I do something like `winget fonts list --name 'Nerd Font'` (display all nerd fonts installed on my system) or `winget fonts list 'Italic'` (list out all italic fonts)?

Reply:
Or also `--version`, `--exact`, etc.

Reply:
I am thinking of creating an index in memory that we can use to better search the installed fonts like the examples you have above.